### ISSUE TITLE
The return of the NaN

### ISSUE THREAD

**Issue by jackbaker1001** [XanaduAI/GradDFT#72]
I've noticed that for a few basis sets, NaN gradients are appearing again when trained using the DIIS SCF loops but not the linear mixing loops. 

I think this is likely because degenerate eigenvectors/eigenvalues are being encountered in the calls to `jnp.linalg.eigh` routines. We can try switching these out for our custom `safe_eigh` which will hopefully fix the bug.

**Comment by jackbaker1001** [comment]
@PabloAMC I still get NaNs on this branch running the self-consistent training in the notebook `~examples/intermediate_notebooks/training_methods_03.ipynb` when switching the basis sets for all molecules to `6-31G`.

Can you tell me if this happens in your local install please?

![Screenshot 2023-09-25 at 2 46 32 PM](https://github.com/XanaduAI/GradDFT/assets/20515391/eed70a21-d222-4e7f-9ab8-769a6f668aac)

**Comment by PabloAMC** [comment]
Did lowering the clipping constant help? I can't replicate the issue on my end with the current one.
